src/utils/config.py

```diff
@@ -0,0 +1,246 @@
+"""Simple utils to assist the config."""
+import pathlib
+import os
+from copy import deepcopy
+from typing import Union
+
+import yaml
+
+from src.utils.defence import (
+    _type_defence, 
+    _validate_str_is_path, 
+    _validate_file_extension
+)
+
+def safeload_yaml(path: Union[str, pathlib.Path]) -> dict:
+    """Load a .yaml file from a path.
+
+    Args:
+        path (Union[str, pathlib.Path]): The path to load the .yaml file from.
+
+    Raises:
+        FileNotFoundError: Raised if there is no file at the given path.
+        TypeError: Raised if the file does not have the .yaml extension.
+
+    Returns:
+        dict: The loaded yaml file as as dictionary.
+    """
+    if not os.path.exists(path):
+        raise FileNotFoundError(
+            f"Attempted to load yaml at: {path}. File "
+            "does not exist."
+        )
+    ext = os.path.splitext(path)[1]
+    if ext != ".yaml":
+        raise TypeError(
+            f"Expected a .yaml file. Got {ext}"
+        )
+    with open(path, "r") as f:
+        yaml_dict = yaml.safe_load(f)
+    return yaml_dict
+
+
+def merge_configs(config1: dict, config2: dict) -> dict:
+    """Merge two config files.
+
+    Args:
+        config1 (dict): The first config file.
+        config2 (dict): The second config file.
+
+    Raises:
+        ValueError: Raised if there are overlapping keys in the configs.
+
+    Returns:
+        dict: The merged config file.
+    """
+    for key in config1:
+        if key in config2:
+            if (
+                isinstance(config2[key], dict) and 
+                isinstance(config1[key], dict)
+            ):
+                for subkey in config1[key]:
+                    if subkey not in config2[key].keys():
+                        config2[key][subkey] = config1[key][subkey]
+                    else:
+                        raise ValueError(f"Overlapping keys in configs: {key}:{subkey}")
+            else:
+                raise ValueError(f"Overlapping keys in configs: {key}")
+
+        else:
+            config2[key] = config1[key]
+    return deepcopy(config2)
+
+
+def _check_has_schema(value: dict) -> bool:
```

Comment:
could we have more descriptive variable names here? It would make it more easy to follow. For example, would it make sense for value to be named "sch_dict" or something else _dict? I tend to think of value as a single value. Also, instead of item, could it be something like "conf_key" ?